pkg/build/registry/buildconfig/strategy.go

```diff
@@ -98,3 +99,28 @@ func dropUnknownTriggers(bc *api.BuildConfig) {
 	}
 	bc.Spec.Triggers = triggers
 }
+
+type statusStrategy struct {
+	strategy
+}
+
+// StatusStrategy is the default logic that applies when updating BuildConfig status
+var StatusStrategy = statusStrategy{Strategy}
+
+// PrepareForUpdate ensures that a BuildConfig spec remains unchanged with the exception of Triggers
+// The LastImageID can be updated with a status change. No other change of the spec is allowed.
+func (statusStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newBC := obj.(*api.BuildConfig)
+	oldBC := old.(*api.BuildConfig)
+
+	newTriggers := newBC.Spec.Triggers
+	newBC.Spec = oldBC.Spec
+	// Allow changes to triggers for LastImageID. If anything else in triggers is
+	// changed, validation of the update will fail.
+	newBC.Spec.Triggers = newTriggers
```

Comment:
This is because we're keeping status inside the trigger spec (LastTriggeredImageID). I'm all in favor of moving it to the status field. Thoughts?